### ISSUE TITLE
FacetWP compatibility

### ISSUE THREAD

**Issue by graylaurenm** [pluginkollektiv/antispam-bee#357]
We're running into a conflict where line 1145 of antispam_bee.php uses ob_start which is never ended and interferes with other output buffers, specifically that of FacetWP in this case.

**Comment by Zodiac1978** [comment]
Hi @graylaurenm  and thank your for this issue!

After looking at https://www.php.net/manual/de/function.ob-start.php and our code, I think you are right and this is a bug.

To speed things up: Is there an easy way I can reproduce the issue which you can share with us?

**Comment by Zodiac1978** [comment]
There is a ticket about best practices:
https://core.trac.wordpress.org/ticket/43258

On this Github readme are some examples mentioned:
https://github.com/nextend/wp-ob-plugins-themes/blob/master/README.md

And a testing plugin:
https://wordpress.org/plugins/output-buffer-tester/

**Comment by graylaurenm** [comment]
@Zodiac1978 I can provide you with a staging site. I will email you those credentials.

**Comment by graylaurenm** [comment]
Sent to the address I found on your website. Subject line is `Staging site for #357` 😄

**Comment by Zodiac1978** [comment]
I got the mail, but not the wordpress credentials.

Can you describe how I can see the bug in action? What steps do I need to do to see them?

**Comment by graylaurenm** [comment]
Sent another email with the credentials included.

To reproduce, you'll:
1. Visit the page we set up with FacetWP
2. Expand one of the drop-downs on the right-hand side
3. Select any of those options (You'll notice that you get an error that FacetWP is unable to find the query.)
4. Next, deactivate Antispam Bee and repeat steps 1-3; the filtering works as expected

**Comment by Zodiac1978** [comment]
I have logged in and saw wpDiscuz active. This is replacing the native comment form completely which makes Antispam Bee mostly useless. 

I still think you are right about not ending the `ob_start` but at the moment ASB is not doing the best because our honeypot is not used. Maybe the other checks are still working but we do net support/test with wpDiscuz and other Tools like that (e.g. Jetpack Comments).

I am wondering why this works without marking all comments as spam.

Additionally I can't reproduce the problem on the mentioned page. I set the filter and it correctly filters the recipes. Although ASB is active. But I see some JS errors in the console.

Have you already fixed this manually or why is this not reproducible for me?

**Comment by graylaurenm** [comment]
I apologize for the delay, I needed to investigate replication but the staging site had been reset. I'm also seeing this work, even with Antispam Bee active. Perhaps it had to do with one of the [two most recent updates](https://facetwp.com/documentation/changelog/) to FacetWP? (I had emailed their support previously.)  Either way, I think we can close this issue out since it's specifically about FacetWP, not the `ob_start`.
